### DISCUSSION TITLE
How to run encoding using multiple cores?

### DISCUSSION THREAD

**Discussion by kundanshrivastava** [PyAV-Org/PyAV#1015] [1. Help]
## Overview

Unable to run PyAV container encoding on multiple CPU's

## Expected behavior

by adding flags:
 self.stream.thread_count = 4
 self.stream.thread_type = 3
Its should work on 4 thread but i don't see that is working as performance not improved at all

## Actual behavior
Encoding should start working on multicore passed by  self.stream.thread_count = 4 and  self.stream.thread_type = 3 arguments.

## Research

I have done the following:
Inside ___init__():
 self.container = av.open(path, mode="w")
 self.stream = self.container.add_stream("h264", rate=round(frame_rate))
 self.stream.pix_fmt = "yuv420p"
 self.stream.thread_count = 4
 self.stream.thread_type = 3
 self.stream.bit_rate = bit_rate

Adding data to container:
#type of src is numpy array
           frame = av.VideoFrame.from_ndarray(src, format="rgba")
           self.container.mux(self.stream.encode(frame))

## Additional context
I am writing the webm video form numpy array data and its working fine but its too slow, therefor exploring the multi core option

**Comment by kundanshrivastava** [comment]
@mikeboers @jlaine @markreidvfx Guys will you please help me here, thanks in Advance

**Comment by phineas-pta** [comment]
`thread_count = 0` should be use as many threads as possible ? did you make it work ?

**Comment by LSKhappychild** [comment]
Any progress on this?
